vm/src/vm.rs

```diff
@@ -399,7 +399,7 @@ impl VirtualMachine {
     }
 
     #[cfg(feature = "rustpython-compiler")]
-    pub fn new_syntax_error(&self, error: &CompileError) -> PyObjectRef {
+    pub fn new_syntax_error(&self, error: &CompileError, filename: Option<&str>) -> PyObjectRef {
```

Comment:
As an idea, add the field `filename: Option<String>` to `CompileError`. This save a lot of passing around the filename. Could this work?

Reply:
That sounds better. I will try it.